tests/cloner_test.go

```diff
@@ -75,6 +76,219 @@ var _ = Describe("[rfe_id:1277][crit:high][vendor:cnv-qe@redhat.com][level:compo
+
+		By("Cloning from the source DataVolume to target1")
+		targetDv1 = utils.NewDataVolumeForImageCloning("target-dv1", "500Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv1)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for clone to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, targetDv1.Name)
+		Expect(err).ToNot(HaveOccurred())
+		matchFile := filepath.Join(testBaseDir, "disk.img")
+		Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(targetDv1), matchFile, md5sum[:32])).To(BeTrue())
+		_, err = utils.WaitPodDeleted(f.K8sClient, "verify-pvc-md5", f.Namespace.Name, verifyPodDeletedTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Cloning from the source DataVolume to target2")
+		targetDv2 = utils.NewDataVolumeForImageCloning("target-dv2", "500Mi", f.Namespace.Name, sourceDv.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv2)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for clone to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, targetDv2.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(targetDv2), matchFile, md5sum[:32])).To(BeTrue())
+		_, err = utils.WaitPodDeleted(f.K8sClient, "verify-pvc-md5", f.Namespace.Name, verifyPodDeletedTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Cloning from the target1 DataVolume to target3")
+		targetDv3 = utils.NewDataVolumeForImageCloning("target-dv3", "500Mi", f.Namespace.Name, targetDv1.Name)
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDv3)
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for clone to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, targetDv3.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(f.VerifyTargetPVCContentMD5(f.Namespace, utils.PersistentVolumeClaimFromDataVolume(targetDv3), matchFile, md5sum[:32])).To(BeTrue())
+	})
+})
+
+var _ = Describe("Validate Data Volume clone to smaller size", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+	tinyCoreIsoURL := fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs)
+
+	var (
+		sourceDv, targetDv *cdiv1.DataVolume
+		err                error
+	)
+
+	AfterEach(func() {
+		if sourceDv != nil {
+			By("Cleaning up source DV")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, sourceDv.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if targetDv != nil {
+			By("Cleaning up target DV")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, targetDv.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Should not allow cloning into a smaller sized data volume", func() {
```

Comment:
Please add the Polarion ID as follows:
It("[rfe_id:CNV-1126][case_id:CNV-1896][crit:High][vendor:cnv-qe@redhat.com][level:component] Should not allow cloning into a smaller sized data volume", func() {

Reply:
Done